synapse/config/server.py

```diff
@@ -904,25 +934,15 @@ def parse_listener_def(num: int, listener: Any) -> ListenerConfig:
         raise ConfigError(DIRECT_TCP_ERROR, ("listeners", str(num), "type"))
 
     port = listener.get("port")
-    if type(port) is not int:
+    socket_path = listener.get("path")
+    # Either a port or a path should be declared at a minimum. Using both would be bad.
```

Comment:
we don't seem to check that one or the other is actually specified (except with an `assert` later, but this should probably be a configerror instead)

Reply:
They...are? Like, three lines down? I don't follow

Reply:
missing `if port is None and socket_path is None: raise ConfigError(..)` I thought

Reply:
Ah. Was fixed up in follow up commit. I think it says in the commit message.

Reply:
Oh I think I see what you mean now, they are both checked for values that are valid and both are checked for existing at the same time but not checked for neither existing. I agree these needs tending to. It's making this part messy though, any ideas to cover all these bases that aren't so...ugly? Again for context, this is before it get stuffed into any kind of `ListenerConfig` and is still fresh from the yaml.

Reply:
Fixed in [d1742a4](https://github.com/matrix-org/synapse/pull/15353/commits/d1742a480c610f39509c80e9709f33c9db2ea0c3)